### PR TITLE
Reorganize the hierarchy on state documentation

### PR DESCRIPTION
**Pull request by AaronFriel**
Changes headings for brevity, moves the service architecture section, and reduces heading depth overall to put more items on the right hand navbar and break up the very large "Logging In" section into more navigable parts.

### CODE REVIEW

File: themes/default/content/docs/intro/concepts/state.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The default experience is to use the hosted Pulumi Service, which takes care of
 
 > Pulumi state does not include your cloud credentials. Credentials are kept local to your client &mdash; wherever the CLI runs &mdash; even when using the managed Pulumi Service backend. Pulumi _does_ store configuration and secrets, but encrypts those secrets using your chosen encryption provider. To learn more, see [Configuration and Secrets]({{< relref "secrets" >}}).
 
-## Backends
+## Deciding On a Backend
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:24]:
```suggestion
## Deciding On a State Backend
```

File: themes/default/content/docs/intro/concepts/state.md [hunk 2]
```diff
@@ -220,13 +227,13 @@ $ pulumi login gs://<my-pulumi-state-bucket>
...
 
-### Logging Out
+## Logging Out
 
 To log out from your currently chosen backend, run the [`pulumi logout` CLI command]({{< relref "/docs/reference/cli/pulumi_logout" >}}). This will remove all credentials information from `~/.pulumi/credentials.json` and you will need to log in again before performing any subsequent stack or state operations.
 
 To change backends, run `pulumi logout` followed by `pulumi login` with the desired backend (or just leave it blank for the default).
 
-### Migrating Between Backends
+## Migrating Between Backends
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:236]:
```suggestion
## Migrating Between State Backends
```

File: themes/default/content/docs/intro/concepts/state.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ The default experience is to use the hosted Pulumi Service, which takes care of
 
 > Pulumi state does not include your cloud credentials. Credentials are kept local to your client &mdash; wherever the CLI runs &mdash; even when using the managed Pulumi Service backend. Pulumi _does_ store configuration and secrets, but encrypts those secrets using your chosen encryption provider. To learn more, see [Configuration and Secrets]({{< relref "secrets" >}}).
 
-## Backends
+## Deciding On a Backend
 
 Pulumi supports two classes of  _backends_ for storing your infrastructure state:
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:26]:
```suggestion
Pulumi supports two classes of state backends for storing your infrastructure state:
```

File: themes/default/content/docs/intro/concepts/state.md [hunk 4]
```diff
@@ -46,9 +44,9 @@ The Pulumi Service backend requires no additional configuration after [installin
...
+Pulumi also lets you manage state yourself using a self-managed backend. Your state is stored as simple JSON files in AWS S3, Azure Blob Store, Google Cloud Storage, an alternative AWS S3 API compatible server such as Minio or Ceph, or on your local filesystem. These self-managed backends are all open source and free to use in any setting. Using a self-managed backend trades off some amount of reliability for additional control over where metadata is stored. For instance, you will need to manually configure secure access, encryption, and history, and devise your own concurrency control and recovery capabilities. To choose a self-managed backend, use the `pulumi login` command [as documented below](#using-a-self-managed-backend).
 
-### Logging In
+## Logging In
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:49]:
```suggestion
## Logging into a State Backend
```

File: themes/default/content/docs/intro/concepts/state.md [hunk 5]
```diff
@@ -95,7 +86,7 @@ User: <your-username>
 Backend URL: https://app.pulumi.com/<your-username>
 ```
 
-#### Logging Into the Pulumi Service Backend
+## Using the Pulumi Service Backend
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:89]:
```suggestion
## Pulumi Service Backend
```

File: themes/default/content/docs/intro/concepts/state.md [hunk 6]
```diff
@@ -118,7 +109,7 @@ To view your access tokens, or create a new one manually, view the <a href="http
 
 <img src="/images/docs/reference/state_tokens.png" alt="Pulumi.com Tokens Page" class="img-bordered">
 
-##### Logging Into a Self-Hosted Pulumi Service Backend
+### Self-Hosted Pulumi Service
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:112]: honestly i think this doesnt need its own heading, and the content could just be at the end of the previous section

File: themes/default/content/docs/intro/concepts/state.md [hunk 7]
```diff
@@ -220,13 +227,13 @@ $ pulumi login gs://<my-pulumi-state-bucket>
 
 To configure credentials for this backend, see [Application Default Credentials](https://cloud.google.com/docs/authentication/production). For additional configuration options, see [GCP Setup]({{< relref "/registry/packages/gcp/installation-configuration" >}}). If you're new to Google Cloud Storage, see [the Google Cloud documentation](https://cloud.google.com/storage/docs/quickstarts).
 
-### Logging Out
+## Logging Out
```
Comment (susanev) [inline themes/default/content/docs/intro/concepts/state.md:230]: can we include this as part of the logging in section instead just at the end of that section?

File: themes/default/content/docs/intro/concepts/state.md (hunk 4 above)
Comment (susanev) [reply themes/default/content/docs/intro/concepts/state.md:49]: if you move the logging out to this same section, we might want to adjust the heading a bit

**Review by susanev — APPROVED** [review]
only take what you need!
thank you!!!!!

File: themes/default/content/docs/intro/concepts/state.md (hunk 7 above)
Comment (AaronFriel) [reply themes/default/content/docs/intro/concepts/state.md:230]: Yeah, that makes sense.

File: themes/default/content/docs/intro/concepts/state.md (hunk 4 above)
Comment (AaronFriel) [reply themes/default/content/docs/intro/concepts/state.md:49]: How about "Logging into and out of State Backends"? Seems a bit wordy.

Comment (susanev) [reply themes/default/content/docs/intro/concepts/state.md:49]: nah i think thats fine

**Comment by susanev** [comment]
@AaronFriel we may want to check that no other pages were linking to the anchors on these pages? or if they are we should update them

**Comment by susanev** [comment]
this page links to the logging in section: https://www.pulumi.com/product/pulumi-service/ i know that for sure, but you might want to search for each anchor link you changed

**Comment by AaronFriel** [comment]
@susanev could you give the page a final look over? 

https://pulumi-hugo-origin-pr-2033-fa18120d.s3-website.us-west-2.amazonaws.com/docs/intro/concepts/state

**Review by susanev — APPROVED** [review]
great work!

:shipit:
